feat(userasterlayers): implement permanent_only truck restrictions

KhaledSakr opened the pull request:
BREAKING CHANGE: this changes the default behavior to be permanent_only instead of active_and_inactive

## What the PR Includes
- [x] implement permanent_only truck restrictions.

jlabeit commented on src/HEREMap.tsx:
```diff
@@ -205,26 +194,29 @@ export const HEREMap = forwardRef<HEREMapRef, HEREMapProps>(({
+
+      const service = platform.getTrafficService();
+      // @ts-ignore
+      const provider = new H.service.traffic.flow.Provider(service);
```
The updated traffic flow only is rather hard to see and is only displayed when zooming in very closely. I would assume this is an issue for the users. 

 Here is a side by side comparison (this pr on the left, old version on the right). 

![Peek 2022-09-07 13-17](https://user-images.githubusercontent.com/2670732/188866240-a1d82685-807d-424c-9c4c-f5ec87b29611.gif)

I am not sure how but ideally we adapt the traffic flow overlay to show up earlier (lower zoom levels) and in general more prominent (e.g. thicker line styling).

KhaledSakr replied:
I made two changes to get similar behavior with the new version:
1. Switched to vector base layer when using traffic, because it supports changing the colors for roads/highways when traffic is enabled
2. Switched to raster traffic layer because it works better with far zoom levels
I added comments to https://github.com/impargo/react-here-maps/pull/14/commits/ba20a1a86838b83b200a3755e4b765cb6395744c to describe the changes

KhaledSakr replied:
The https://github.com/impargo/impargo-apps/pull/7966 branch is also updated with these changes

jlabeit reviewed (commented):
Code wise this looks pretty good. I also tested it a little bit. One thing I've noticed is regarding the traffic flow. The rest looks quite good I would say.

jlabeit reviewed (approved):
Nice! 
I don't have any further comments :rocket:

KhaledSakr committed:
feat(userasterlayers): implement permanent_only truck restrictions

BREAKING CHANGE: this changes the default behavior to be
permanent_only instead of active_and_inactive

KhaledSakr commented:
/publish

david-paulus-saad commented:
RUN URL: https://github.com/impargo/react-here-maps/actions/runs/12926669338
PACKAGE VERSION: 3.0.0-pre-25
